status-im/status-legacy
# Mobile UI for desktop

fixes #7399

### Summary:

Mobile UI used for desktop platform

### Review notes (optional):
Most issues solved on `react-native-desktop` side and with the help of `(if desktop?...)`.
But there is one area that needs special attention during review:
On mobile `basic-text-input` component saves entered text in re-frame database. Whenever input changes, component recreates. That doesn't cause issues because typing speed on mobile is much slower.
But on desktop typing speed significantly faster and re-rendering causes characters loss. So I had to create component `basic-text-input-desktop` and save input text inside component state instead of re-frame database. Drawback of that decision - typed text doesn't save when I close chat and reopen it again. 
Would appreciate your help in solving this.

### Testing notes (optional):
I believe mobile builds should be tested especially careful because problems in desktop can be solved in a separate issues, but we don't want any regressions on mobile.

#### Platforms (optional)
- Android
- iOS
- macOS
- Linux
- Windows


#### Areas that maybe impacted (optional)
**Functional**
- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

**Non-functional**
- battery performance
- CPU performance / speed of the app
- network consumption


### Steps to test:
- Open Status
- check everything..




status: ready
